src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -267,70 +264,114 @@ impl StaticHeader {
     /// Return the latest copy that validates as a Stratis BDA, however verify both
     /// copies and if one validates but one does not, re-write the one that is incorrect.  If both
     /// copies are valid, but one is newer than the other, rewrite the older one to match.
-    /// Return None if the static header's magic does not match for *both* copies.
+    /// Return None if it's not a Stratis device.
+    /// Return an Err if the metadata seems to indicate that the device is
+    /// a Stratis device, but no well-formed signature block could be read.
```

Comment:
This returns an error in two situations:
* If the metadata seems to indicate that the device is a Stratis deivce, but no well-formed signature could be found.
* If a well-formed signature was found but an error occurred during the repair of a second, ill-formed, old, or unreadable signature.

I think it is right to return an error in the first case, but I think the second should possibly be handled differently. I think a good option is to enrich the return type to
```StratisResult<Option<StaticHeader>, Option<some error thing if repair fails>>```.
Alternatively, it is fine to open an issue to address this later.

Reply:
New issue: https://github.com/stratis-storage/stratisd/issues/1443